tools/generator/autorest/autorest.go

```diff
@@ -130,14 +130,14 @@ func (g *Generator) StderrPipe() (io.ReadCloser, error) {
 
 // GenerateError ...
 type GenerateError struct {
-	Arguments []model.Option
-	Message   string
+	Options model.Options
+	Message string
 }
 
 // Error ...
 func (e *GenerateError) Error() string {
-	arguments := make([]string, len(e.Arguments))
-	for i, o := range e.Arguments {
+	arguments := make([]string, len(e.Options.Arguments()))
+	for i, o := range e.Options.Arguments() {
```

Comment:
Is it better to use arguments defined above?
for i, o := range arguments {

Reply:
We are iterating over the `[]model.Option` here and accumulate them to a string slice. 
I allocate the slice `arguments` with a specific length for performance consideration (which matters very little)